addons/binding/org.openhab.binding.yamahareceiver2/src/main/java/org/openhab/binding/yamahareceiver2/handler/YamahaReceiverHandler.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver2.handler;
+
+import java.io.IOException;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.yamahareceiver2.YamahaReceiverBindingConstants;
+import org.openhab.binding.yamahareceiver2.discovery.ZoneDiscoveryService;
+import org.openhab.binding.yamahareceiver2.internal.YamahaReceiverState;
+import org.openhab.binding.yamahareceiver2.internal.protocol.YamahaReceiverCommunication;
+import org.openhab.binding.yamahareceiver2.internal.protocol.YamahaReceiverCommunication.Zone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link YamahaReceiverHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author David Gräff - Initial contribution
+ */
+public class YamahaReceiverHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(YamahaReceiverHandler.class);
+    private String host;
+    private int refrehInterval = 0;
+    private YamahaReceiverState state = null;
+    private Timer refreshTimer;
+    private ZoneDiscoveryService zoneDiscoveryService;
+
+    public YamahaReceiverHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * We handle updates of this thing ourself.
+     */
+    @Override
+    protected void updateThing(Thing thing) {
```

Comment:
This must not be overridden - this method is called by the binding to inform the framework about updates, not the other way round!